net/FlatBuffers/ByteBuffer.cs

```diff
@@ -38,12 +40,12 @@ namespace FlatBuffers
     /// </summary>
     public class ByteBuffer
     {
-        private readonly byte[] _buffer;
+        protected byte[] _buffer;
         private int _pos;  // Must track start of the buffer.
 
-        public int Length { get { return _buffer.Length; } }
+        public virtual int Length { get { return _buffer.Length; } }
 
-        public byte[] Data { get { return _buffer; } }
```

Comment:
Why is `Data` removed? was this unused?

Reply:
`Data` was used, mainly for length checks and memory copies. I thought that fully hiding the underlying buffer was slightly better design, especially when dealing with slices. Thus, the buffer's content is modified/read using the `Put*` and `Get*` methods.